crates/typos-lsp/src/typos.rs

```diff
@@ -46,10 +53,41 @@ impl Instance<'_> {
         let ignore = ignores.build()?;
 
         Ok(Instance {
+            project_root: path.to_path_buf(),
             ignores: ignore,
             engine,
         })
     }
+
+    /// Returns the typos_cli configuration files that are relevant for the given path. Note that
+    /// all config files are read by typos_cli, and the settings are applied in precedence order:
+    ///
+    /// <https://github.com/crate-ci/typos/blob/master/docs/reference.md>
+    pub fn config_files_in_project(
```

Comment:
I see - this is totally reasonable. I think I misunderstood how the config files are "inherited".

I will change the logic to what you proposed in 1 and 2 👍🏻

Reply:
Ok, now the typo can be ignored in two places:

- in the project's configuration file
- in the explicit configuration file

<img width="854" alt="image" src="https://github.com/tekumara/typos-lsp/assets/300791/9499723e-411b-4a5a-9244-c1001cd26116">

However, it looks like if I remove the project's configuration file, the root is resolved to my home directory. This causes "ignore `foo` in the project" to add the ignore to that file instead.

🤔 I'm not sure what causes this. I do have that file in my home directory, so maybe this is some weirdness in the config resolution logic.

<img width="1716" alt="image" src="https://github.com/tekumara/typos-lsp/assets/300791/8c2c7f64-4c63-409a-9daa-9acaf501f56c">

Any idea what could be a more reliable way to resolve the project directory?

Reply:
Are you using nvim-lspconfig? It's [root_dir logic is](https://github.com/neovim/nvim-lspconfig/blob/056f569f71e4b726323b799b9cfacc53653bceb3/lua/lspconfig/server_configurations/typos_lsp.lua#L6): 

```lua
root_dir = util.root_pattern('typos.toml', '_typos.toml', '.typos.toml'),
```

ie: it will include ancestors in its search for config files